services/streammanager/wunderkind/wunderkindmanager_test.go

```diff
@@ -0,0 +1,136 @@
+package wunderkind
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger/mock_logger"
+	mock_lambda "github.com/rudderlabs/rudder-server/mocks/services/streammanager/lambda"
+)
+
+var (
+	sampleMessage    = "sample payload"
+	sampleFunction   = "sampleLambdaFunction"
+	sampleExternalID = "sampleExternalID"
+	sampleIAMRoleARN = "sampleRoleArn"
+	invocationType   = "RequestResponse"
+)
+
+func TestNewProducer(t *testing.T) {
+	conf := config.New()
+	conf.Set("WUNDERKIND_REGION", "us-east-1")
+	conf.Set("WUNDERKIND_IAM_ROLE_ARN", sampleIAMRoleARN)
+	conf.Set("WUNDERKIND_EXTERNAL_ID", sampleExternalID)
+	conf.Set("WUNDERKIND_LAMBDA", sampleFunction)
+
+	producer, err := NewProducer(conf)
+	require.Nil(t, err)
+	require.NotNil(t, producer)
+	require.NotNil(t, producer.client)
+
+	// Invalid Region
+	conf.Set("WUNDERKIND_EXTERNAL_ID", "")
+	producer, err = NewProducer(conf)
+	require.Nil(t, producer)
+	require.Equal(t, "creating session: externalID is required for IAM role", err.Error())
+}
```

Comment:
We can create two independent test case here.
```suggestion
func TestNewProducer(t *testing.T) {
	t.Run("valid", func(t *testing.T) {
		conf := config.New()
		conf.Set("WUNDERKIND_REGION", "us-east-1")
		conf.Set("WUNDERKIND_IAM_ROLE_ARN", sampleIAMRoleARN)
		conf.Set("WUNDERKIND_EXTERNAL_ID", sampleExternalID)
		conf.Set("WUNDERKIND_LAMBDA", sampleFunction)

		producer, err := NewProducer(conf)
		require.Nil(t, err)
		require.NotNil(t, producer)
		require.NotNil(t, producer.client)
	})
	t.Run("invalid region", func(t *testing.T) {
		conf := config.New()
		conf.Set("WUNDERKIND_EXTERNAL_ID", "")
		producer, err := NewProducer(conf)
		require.Nil(t, producer)
		require.Equal(t, "creating session: externalID is required for IAM role", err.Error())
	})
}
```

Reply:
addressed